src/Evolution/DiscontinuousGalerkin/Actions/ComputeTimeDerivative.hpp

```diff
@@ -470,6 +480,12 @@ ComputeTimeDerivative<Metavariables>::apply(
         make_not_null(&box));
   }
 
+  if constexpr (Metavariables::local_time_stepping and
+                not SelfStart::is_self_start_procedure<ActionList>) {
```

Comment:
I don't love the `is_self_start_pro...` because subcell will likely wreak a lot of havoc in self starting (this was the case in my previous implementation where I never got it far enough to actually fully work, though I'm hopeful I can simplify it a lot this time around). Couldn't this check be done via runtime information from either the time stepper and/or the slab? E.g. I thought all slabs during self-start are before (in the sense of the direction that time flows) the initial time. That seems a lot less likely to break and causes less subtle coupling between pieces of code that might check if they are in self-start. I also don't think you can do LTS until a certain amount of time after self-start because the history is non-monotonic, but @wthrowe will need to say what the restrictions are there.

(I haven't looked at the rest of the code, I just wanted to see the changes in ComputeTimeDerivative)

Reply:
The sanctioned method of determining you are in self-start is checking if the slab number is negative, but I'm not sure why you want to conditionally take a step anyway.

Reply:
Here, the point is that we can only make the stepping a subroutine of the `ComputeTimeDerivative` during the main evolution and not during self-start, because self-start needs to replace the `RecordTimeStepperData` with it's own looping mechanism. During self-start, the action list explicitly contains the `UpdateU` and `RecordTimeStepperData` actions.

Reply:
I also don't like that this adds _another_ level of nested conditionals into the Metavariables. I'd generally prefer going the other direction. Maybe we can discuss this at the call today (Wed. Feb. 10) if both @wthrowe and @moxcodes can make it. DG-subcell needs even more jumping around in the action list (roughly), and making the action list conditional on self-starting will probably mean I have to rip out the error adjustment to get DG-subcell to work.